### PR TITLE
SOV-3009: voting actions

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-3009

### CODE REVIEW

File: apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    'protocol',
+    defaultChainId,
+  );
+
+  const { balance, loading } = useAssetBalance(SupportedTokens.rbtc);
+  const status = useProposalStatus(proposal);
+
+  const isDisabled = useMemo(() => {
+    return !account || decimalic(balance).eq('0') || !contract || loading;
+  }, [account, balance, loading, contract]);
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx]:
```suggestion
  const isDisabled = useMemo(
    () => !account || balance.isZero() || !contract || loading,
    [account, balance, loading, contract],
  );
```

File: apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    return status === ProposalState.Pending
+      ? ProposalActionType.queue
+      : ProposalActionType.execute;
+  }, [status]);
+
+  const actionButtonText = useMemo(() => {
+    return status === ProposalState.Pending
+      ? t(pageTranslations.actions.queue)
+      : t(pageTranslations.actions.execute);
+  }, [status]);
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx]:
```suggestion
  const isActionVisible = useMemo(
    () =>
      contract &&
      (status === ProposalState.Succeeded || status === ProposalState.Pending),
    [status, contract],
  );

  const action = useMemo(
    () =>
      status === ProposalState.Pending
        ? ProposalActionType.queue
        : ProposalActionType.execute,
    [status],
  );

  const actionButtonText = useMemo(
    () =>
      status === ProposalState.Pending
        ? t(pageTranslations.actions.queue)
        : t(pageTranslations.actions.execute),
    [status],
  );
```

File: apps/frontend/src/locales/en/translations.json [hunk 3]
```diff
@@ -666,7 +666,11 @@
         "reject": "Reject",
         "rejectText": "You voted to reject this proposal",
         "noVote": "No vote cast on this proposal",
-        "ineligibleVote": "Ineligible to vote - no SOV staked at proposal creation time <br/> Stake SOV to be eligible to vote on future proposals"
+        "ineligibleVote": "Ineligible to vote - no SOV staked at proposal creation time <br/> Stake SOV to be eligible to vote on future proposals",
```
Comment (tiltom) [inline apps/frontend/src/locales/en/translations.json]: You shouldn't have HTML tags in translations - break it into multiple translations instead and style it in the component.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: apps/frontend/src/locales/en/translations.json (hunk 3 above)
Comment (pietro-maximoff) [reply apps/frontend/src/locales/en/translations.json]: I didn't add it, but I will fix it

**Review by tiltom — APPROVED** [review]

File: apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    () =>
+      contract &&
+      (status === ProposalState.Succeeded || status === ProposalState.Pending),
+    [status, contract],
+  );
+  const action = useMemo(
+    () =>
+      status === ProposalState.Pending
+        ? ProposalActionType.queue
+        : ProposalActionType.execute,
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx]: Condition is wrong here - you can queue proposal thats successed and only executed queued proposal:

```suggestion
      switch(status) {
        case ProposalState.Succeeded:
          return ProposalActionType.queue;
        case ProposalState.Queued:
          return ProposalActionType.execute;
      }
```

File: apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    [status],
+  );
+  const actionButtonText = useMemo(
+    () =>
+      status === ProposalState.Pending
+        ? t(pageTranslations.actions.queue)
+        : t(pageTranslations.actions.execute),
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx]: Compare `ProposalActionType` instead `ProposalState` to show translations

File: apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+  const isDisabled = useMemo(
+    () => !account || balance.isZero() || !contract || loading,
+    [account, balance, loading, contract],
+  );
+
+  const isActionVisible = useMemo(
+    () =>
+      contract &&
+      (status === ProposalState.Succeeded || status === ProposalState.Pending),
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx]:
```suggestion
      (status === ProposalState.Succeeded || status === ProposalState.Queued),
```
`ProposalState.Pending` - is a state just after proposal is created, it's pending to be voted (only valid for 1 block)

File: apps/frontend/src/constants/gasLimits.ts [hunk 7]
```diff
@@ -24,4 +24,5 @@ export const GAS_LIMIT = {
   BORROW: 1_500_000,
   REPAY_LOAN: 950_000,
   WITHDRAW_LOAN_COLLATERAL: 150_000,
+  PROPOSAL_ACTION: 250_000,
```
Comment (creed-victor) [inline apps/frontend/src/constants/gasLimits.ts]: make different actions for queue and execute, for queing 250k will be enough (maybe even less will work), but for execution we may need full 6M

**Review by creed-victor — CHANGES_REQUESTED** [review]

**Review by creed-victor — APPROVED** [review]

**Pull request merged**